src/uct/tcp/tcp_cm.c

```diff
@@ -75,6 +78,11 @@ void uct_tcp_cm_change_conn_state(uct_tcp_ep_t *ep,
             /* Since ep::peer_addr is 0'ed, we have to print w/o peer's address */
             full_log = 0;
         }
+
+        if (old_conn_state == UCT_TCP_EP_CONN_STATE_CONNECTED) {
+            ucs_assert(ep->tx.unable_do_cnt < UCT_TCP_EP_TX_DO_MAX);
+            ++ep->tx.unable_do_cnt;
```

Comment:
why do we need this counter if already have conn state?

Reply:
This counter is needed to reduce the number of condition checked when need to check whether it is possible to do TX or not.
Also, the counter is needed to decrement it and then check if it is 0 - allow TX. Since we have three conditions, one flag isn’t enough

Reply:
can we check some flags at once with & operations?
the problem with counter is harder to debug - you don't know the state expecilitly

Reply:
> can we check some flags at once with & operations?
> the problem with counter is harder to debug - you don't know the state expecilitly

I've removed this code, it is not needed anymore, since we have 64-bit put_cnt